controllers/application_controller.go

```diff
@@ -165,10 +187,85 @@ func (r *ApplicationReconciler) reconcileHRStatus(application *v1.Application, c
 		v1.AppErrorStatus(application, "updated Helm Release status not available")
 		return errors.New("HelmRelease status is not current")
 	}
-	application.Status.Conditions = hr.GetConditions()
+	for _, condition := range hr.GetConditions() {
+		apimeta.SetStatusCondition(&application.Status.Conditions, condition)
+	}
 	return nil
 }
 
+func (r *ApplicationReconciler) reconcilePodStatus(ctx context.Context, application *v1.Application) error {
+	// Analyze the status of the pods under the HR, if any
+	log.Info("Reconciling pod status", "application", application.Name, "namespace", application.Namespace)
+	// XXX: This is for deployments. We need to make it clear that right now, this feature is "best effort".
+	var deploymentList appsv1.DeploymentList
+	if err := r.List(ctx, &deploymentList, &client.ListOptions{Namespace: application.Namespace}); err != nil {
+		return err
+	}
+	// Try to retrieve the deployment
+	var releaseDeployment *appsv1.Deployment
```

Comment:
There are labels that are set by helm controller that we can search by and not have to get the list and go over the range of deployments. These are the following labels you can search for deployments by:
    helm.toolkit.fluxcd.io/name: 
    helm.toolkit.fluxcd.io/namespace:

Reply:
Awesome catch! I could've sworn only the annotations `meta.helm.sh/release-name` and `meta.helm.sh/release-namespace` were reasonable, but you're right, I can indeed see the two labels.

I'll update it shortly.

Reply:
Resolved by e84ddaa